viv_utils/emulator_drivers.py

```diff
@@ -182,43 +155,41 @@ def doHook(self, pc, op):
                 if r is not None:
                     # take the first result
                     # not ideal, but works in the common case
-                    self._logger.debug("monitor hook handled call: %s", callname)
+                    logger.debug("driver: monitor hook handled call: %s", callname)
                     return True
             except StopEmulation:
                 raise
             except Exception as e:
-                mon.logAnomaly(emu, pc,
-                        "%s.apicall failed: %s" % (mon.__class__.__name__, e))
+                logger.debug("%s.apicall failed: %s", mon.__class__.__name__, e)
 
         for hook in self._hooks:
             try:
                 ret = hook.hook(callname, self, callconv, api, argv)
                 # take the first result
                 # not ideal, but works in the common case
                 if ret is not None:
-                    self._logger.debug("driver hook handled call: %s", callname)
+                    logger.debug("driver: hook handled call: %s", callname)
                     return True
+
+                if callname and callname not in ("UnknownApi", "?"):
+                    logger.debug("driver: hook API call NOT handled: %s", callname)
```

Comment:
when done here this triggers on every hook iteration, even though a later hook may have actually handled the call

Reply:
fixed